data/persistence/src/commonMain/kotlin/social/androiddev/common/persistence/localstorage/DodoAuthStorageImpl.kt

```diff
@@ -20,7 +22,7 @@ internal class DodoAuthStorageImpl(
     private val settings: Settings,
     private val json: Json
 ) : DodoAuthStorage {
-
+    private val lock = Mutex()
```

Comment:
Should this be passed in through the constructor instead? Can always supply a default value in-line as well. We could then mock/fake easily with tests

Reply:
Yes, that's great suggestion!